File: Fractal Benchmark/mandelbrot.rs
use std::time::{SystemTime, UNIX_EPOCH};


const BAILOUT: f64 = 16.0;
const MAX_ITERATIONS: i32 = 1000;


fn iterate (x: f64, y: f64 ) -> i32
{
    let cr = y - 0.5;
    let ci = x;
    let mut zi = 0.0;
    let mut zr = 0.0;
    let mut i = 0;

    while i < MAX_ITERATIONS {
        i += 1;
        let temp = zr * zi;
        let zr2 = zr * zr;
        let zi2 = zi * zi;
        zr = zr2 - zi2 + cr;
        zi = temp + temp + ci;
        if zi2 + zr2 > BAILOUT {
            return i;
        }
    }
    return 0;
}

fn mandelbrot ( runs: i32 )
{
    for _ in 0..runs {
        for y in -39..39 {
            println!();
            for x in -39..39 {
                if iterate(x as f64 / 40.0, y as f64 / 40.0) > 0 {
                    print!(" ");
                } else {
                    print!("*");
                }
            }
        }
        println!();
    }
}

fn main ( )
{
    let args: Vec<String> = std::env::args().collect();

	let start = SystemTime::now();
	let runs = if args.len() <= 1 {
	    1
	} else {
	    args[1].parse::<i32>().unwrap_or(1)
	};

    mandelbrot(runs);

    let end = SystemTime::now();
    let elapsed = end.duration_since(UNIX_EPOCH).unwrap().as_secs_f64()
		- start.duration_since(UNIX_EPOCH).unwrap().as_secs_f64();
    eprintln!("Rust Elapsed {:.2}", elapsed);
}

File: Fractal Benchmark/mandelbrot.go
package main

import (
	"fmt"
	"os"
	"strconv"
	"time"
)

var bailout float64 = 16
var maxIterations = 1000

func main ( ) {
	start := time.Now()
	var runs = 1
	if len(os.Args) > 1 {
		runs, _ = strconv.Atoi(os.Args[1])
	}
	mandelbrot(runs)
	diff := time.Since(start)
	fmt.Fprintf(os.Stderr, "Go Elapsed %.3f\n", diff.Seconds())
	// fmt.Fprintf(os.Stderr, "Go Elapsed %s\n", diff)
}


func iterate ( x float64, y float64 ) int {
	cr := y - 0.5
	ci := x
	zi := 0.0
	zr := 0.0
	i := 0

	for {
		i++
		temp := zr * zi
		zr2 := zr * zr
		zi2 := zi * zi
		zr = zr2 - zi2 + cr
		zi = temp + temp + ci
		if zi2+zr2 > bailout {
			return i
		}
		if i > maxIterations {
			return 0
		}
	}
	return 0
}


func mandelbrot ( runs int ) {
	for i := 0; i < runs; i++ {
		for y := -39; y < 39; y++ {
			fmt.Println("")
			for x := -39; x < 39; x++ {
				if iterate(float64(x)/40.0, float64(y)/40.0) == 0 {
					fmt.Print("*")
				} else {
					fmt.Print(" ")
				}
			}
		}
		fmt.Println("")
	}
}
